### PR TITLE
feat(release): Also bundle dragonfly-debug builds

### PR DESCRIPTION
**Pull request by Pothulapati**

### CODE REVIEW

**Comment by Pothulapati** [comment]
This mostly works but somehow the regression tests keep failing: https://github.com/dragonflydb/dragonfly/actions/runs/7593673964/job/20684393614

@romange Any idea?

**Comment by Pothulapati** [comment]
We seem to get the debug binaries with this change: https://github.com/dragonflydb/dragonfly/actions/runs/7863662234

(Now, Remving all the changes for testing)

File: Makefile [hunk 1]
```diff
@@ -38,6 +38,7 @@ build:
 
 package:
 	cd $(RELEASE_DIR); \
+	tar cvfz $(RELEASE_NAME)-debug.tar.gz dragonfly ../LICENSE.md; \
```
Comment (Pothulapati) [inline Makefile:41]: This is the actual change. Essentially we are storing a -debug.tar.gz wih the dragonfly binary before its stripped of symbols and then expect the release pipeline to also publish these files.

Comment (romange) [reply Makefile:41]: Why there is a whitespace change? Can you please remove unnecessary changes?

Comment (Pothulapati) [reply Makefile:41]: Sure! Linter did its thing!

Comment (Pothulapati) [reply Makefile:41]: Fixed!

Comment (romange) [reply Makefile:41]: ok, just to make sure the naming is not misleading, I suggest that we call it `$(RELEASE_NAME)-dbgsym.tgz`

Comment (Pothulapati) [reply Makefile]: Done!

**Review by romange — APPROVED** [review]

**Pull request merged**